### PR TITLE
Basic Set implementation

### PR DESCRIPTION
**Pull request by Gamerfiend**
Many parts from Map

### CODE REVIEW

**Comment by Gamerfiend** [comment]
Updated this to include set methods, union, intersection, difference

**Comment by bendmorris** [comment]
The "compile-src" tests have moved into tests/functional, so please put your tests there.

It looks like there are some failing tests. If you run them locally do you see them fail?

**Comment by Gamerfiend** [comment]
For the set, or the general library?

File: std/kit/map.kit [hunk 1]
```diff
@@ -6,6 +6,12 @@ trait Hashable {
     function hash(): Int;
 }
```
Comment (bendmorris) [inline std/kit/map.kit]: Since this is shared now, maybe it should go in a separate "kit.hash" module.

Comment (Gamerfiend) [reply std/kit/map.kit]: That’s what I suggested a while back, move the traits and their implementations for standard types to a new file. I can do that :)

File: std/kit/set.kit [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    public function differenceSet(other: Set[T]){
+        for item in other{
+            if this.exists(item){
+                this.remove(item);
+            }
+        }
+    }
+
+     rules {
+        ($this[$k]) => $this.put($k);
```
Comment (bendmorris) [inline std/kit/set.kit]: Seems a little weird to me for array access to insert a value into the set. Maybe this should be `exists` instead. And maybe `+=` would make sense to insert?

Comment (Gamerfiend) [reply std/kit/set.kit]: Oversight on my part... I'll fix that :)

**Pull request closed without merging**